code/controllers/subsystem/starmap.dm

```diff
@@ -41,14 +41,20 @@ SUBSYSTEM_DEF(starmap)
 
 	var/list/stations = list()
 	var/list/wreckages = list()
+	var/list/station_modules = list()
 
 	var/initial_report = 0
 
 /datum/controller/subsystem/starmap/Initialize(timeofday)
-	var/list/resources = typesof(/datum/star_resource) - /datum/star_resource
+	var/list/resources = subtypesof(/datum/star_resource)
 	for(var/i in resources)
 		star_resources += new i
 
+	for(var/i in subtypesof(/datum/station_module))
+		var/datum/station_module/module = new i
+		station_modules[i] = module.rarity
```

Comment:
So I actually found out recently you can use initial() to get uninstanced var values.
`station_modules[i] = initial(module.rarity)`
Just cut out the `new` in the previous line